cli/src/org/partiql/shell/Shell.kt

```diff
@@ -0,0 +1,285 @@
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at:
+ *
+ *      http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package org.partiql.shell
+
+import com.amazon.ion.system.IonTextWriterBuilder
+import com.amazon.ionelement.api.toIonValue
+import com.google.common.base.CharMatcher
+import com.google.common.util.concurrent.Uninterruptibles
+import org.fusesource.jansi.AnsiConsole
+import org.jline.reader.EndOfFileException
+import org.jline.reader.LineReader
+import org.jline.reader.LineReaderBuilder
+import org.jline.reader.UserInterruptException
+import org.jline.reader.impl.completer.NullCompleter
+import org.jline.terminal.TerminalBuilder
+import org.jline.utils.AttributedStringBuilder
+import org.jline.utils.AttributedStyle
+import org.jline.utils.InfoCmp
+import org.joda.time.Duration
+import org.partiql.lang.CompilerPipeline
+import org.partiql.lang.eval.Bindings
+import org.partiql.lang.eval.EvaluationSession
+import org.partiql.lang.eval.ExprValue
+import org.partiql.lang.eval.ExprValueFactory
+import org.partiql.lang.eval.delegate
+import org.partiql.lang.syntax.Parser
+import org.partiql.lang.util.ConfigurableExprValueFormatter
+import org.partiql.lang.util.ExprValueFormatter
+import java.io.Closeable
+import java.io.OutputStream
+import java.io.PrintStream
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.Locale
+import java.util.Properties
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicBoolean
+import javax.annotation.concurrent.GuardedBy
+
+private const val PROMPT_1 = "PartiQL> "
+private const val PROMPT_2 = "   | "
+private const val BAR_1 = "===' "
+private const val BAR_2 = "--- "
+private const val WELCOME_MSG = "Welcome to the PartiQL REPL!"
+
+private const val HELP = """
+!add_to_global_env  Adds a value to the global environment
+!global_env         Displays the current global environment
+!list_commands      Prints this message
+!help               Prints this message
+!history            Prints command history
+!exit               Exits the shell
+!clear              Clears the screen
+"""
+
+private val EXIT_DELAY: Duration = Duration(3000)
+
+/**
+ * Initial work to replace the REPL with JLine3. I have attempted to keep this similar to Repl.kt, but have some
+ * opinions on ways to clean this up in later PRs.
+ */
+class Shell(
+    private val valueFactory: ExprValueFactory,
+    private val output: OutputStream,
+    private val parser: Parser,
+    private val compiler: CompilerPipeline,
+    private val initialGlobal: Bindings<ExprValue>,
```

Comment:
Since this is used to only initialize `globals`, it might not need to be a class member